dotcom-rendering/src/components/Discussion.tsx

```diff
@@ -33,6 +33,10 @@ export type Props = {
 	reportAbuseUnauthenticated: ReturnType<typeof reportAbuse>;
 };
 
+export const DispatchContext = React.createContext<Dispatch<Action> | null>(
+	null,
+);
```

Comment:
Doing this allows you to drop the `null` type, so you avoid having to make checks for `dispatch` being `null` everywhere you use it.

```suggestion
export const DispatchContext = React.createContext<Dispatch<Action>>(
	() => { debug('dotcom', 'The 'dispatch' function in discussion is not defined! Did you remember to include a context provider for it?' }
);
```

This is `debug` from `guardian/libs`, which I think might be a good fit for this case? @mxdvl

Reply:
This works, but should we use the same pattern as [`ConfigContext`](https://github.com/guardian/dotcom-rendering/blob/4b48425e3f7dfa8a96fb1075c1f7a1c02637a5bf/dotcom-rendering/src/components/ConfigContext.tsx#L20) instead?